core/server/api/canary/identities.js

```diff
@@ -0,0 +1,36 @@
+const settings = require('../../services/settings/cache');
+const urlUtils = require('../../lib/url-utils');
+const jwt = require('jsonwebtoken');
+const jose = require('node-jose');
+const issuer = urlUtils.urlFor('admin', true);
+
+const dangerousPrivateKey = settings.get('ghost_private_key');
+const keyStore = jose.JWK.createKeyStore();
+const keyStoreReady = keyStore.add(dangerousPrivateKey, 'pem');
+
+const getKeyID = async () => {
+    const key = await keyStoreReady;
+    return key.kid;
+};
+
+const sign = async (claims, options) => {
+    const kid = await getKeyID();
+    return jwt.sign(claims, dangerousPrivateKey, Object.assign({
+        issuer,
+        expiresIn: '30m',
+        audience: '@TODO@',
+        algorithm: 'RS256',
+        keyid: kid
+    }, options));
+};
+
+module.exports = {
+    docName: 'identities',
+    permissions: true,
+    read: {
+        permissions: true,
+        async query(frame) {
+            return await sign({sub: frame.user.get('email')});
```

Comment:
@allouis think this is the last bit that needs some discussion. What this does is returns a string from the controller. Usually we return objects with properties instead. Don't think we'd need any more properties here (but just in case we need to include the user role in the future) would remake this response into something like:
```js
const token = await sign({sub: frame.user.get('email')});
return {
  token
};
```
Wdyt?

Reply:
> but just in case we need to include the user role in the future

In this case we would add it inside the token!

Having said that - I have no real opinion on the output format, I think you suggested before about trying to stick with json api style?

Reply:
> trying to stick with json api style?

Yes, this would be the exact intention

Reply:
@allouis moved the signature to a 'token' property. Wdyt about the naming?